### PR TITLE
Multi Language support

### PR DESCRIPTION
**Pull request by LightDestory**
This PR address #205 #191 

I have implemented the basic structure to make it work. I am re-using the `vars.py` file as a container for all the texts and ids.

If you have any good alternative feel free to suggest.

TO-DO:
 - [x] Put all texts inside the dictionary
 - [x] Translate all texts to  English

### CODE REVIEW

File: module/utils/multi_lang_utils.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from ..data.vars import translations
+
+
+def get_locale(language_code: str, text_id: int) -> str:
+    if text_id in translations[language_code]:
+        return translations[language_code][text_id]
+    else:
+        return translations["it"][text_id]
```
Comment (Helias) [inline module/utils/multi_lang_utils.py:8]:
```suggestion
    return translations["it"][text_id]
```

File: module/data/vars.py [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+# Translation Dictionary
+translations: dict[str, dict[int, str]] = {
+    "it": {
+        TEXT_IDS.START_TEXT_ID: "Benvenuto! Questo bot è stato realizzato dagli studenti del Corso di Laurea in Informatica al fine di suppotare gli studenti del DMI!\nPer scoprire cosa puoi fare usa /help o premi il pulsante ❔ Help"
+    },
+    "en": {
+        TEXT_IDS.START_TEXT_ID: "Welcome! This bot is developed by Computer Science's students to help fellow DMI's students!\nUse /help or press the ❔ Help button to know what this bot can do"
+    }
+}
```
Comment (Helias) [inline module/data/vars.py:18]: I don't really like this approach :S
We could create a yaml or json file like `it.json`, `en.json` .
So, if tomorrow wants to make `es.json` it's easy to copy-paste the en.json and translate it.

Comment (Helias) [reply module/data/vars.py:18]: Moreover, it's not important, but we are loosing the possibility to change the .md files while the bot is running, in this way we need to restart the bot to reload the texts

Comment (LightDestory) [reply module/data/vars.py:18]: > Moreover, it's not important, but we are loosing the possibility to change the .md files while the bot is running, in this way we need to restart the bot to reload the texts

I didn't know about this issue. Well, should be base the translation on md files and convert the get_locale into a path resolver? xD

Comment (LightDestory) [reply module/data/vars.py:18]: Hmm maybe we can use json files and hot-read them everytime we need to emit a text?

**Review by Helias — APPROVED** [review]

**Review by drendog — APPROVED** [review]

File: module/data/vars.py [hunk 3]
```diff
@@ -3,70 +3,120 @@
...
+    # /esami
+    EXAMS_SELECT_SESSION_TEXT_ID = auto()
+    EXAMS_SESSION_1_TEXT_ID = auto()
+    EXAMS_SESSION_2_TEXT_ID = auto()
+    EXAMS_SESSION_3_TEXT_ID = auto()
+    EXAMS_SESSION_4_TEXT_ID = auto()
+    EXAMS_USAGE_TEXT_ID = auto()
+
+
+PLACE_HOLDER: str = "%capybara%"
```
Comment (drendog) [inline module/data/vars.py:119]: :heart:

**Review by drendog — APPROVED** [review]

File: module/commands/aulario.py [hunk 4]
```diff
@@ -28,40 +29,41 @@ def aulario(update: Update, context: CallbackContext, chat_id: int = None, messa
...
+    message_id: int = query.message.message_id
 
-    d = data.split("_")
-    direction = d[1]
-    year = int(d[2])
-    month = int(d[3])
-    days = int(d[4])
+    d: list[str] = data.split("_")
+    direction: str = d[1]
+    year: int = int(d[2])
```
Comment (TendTo) [inline module/commands/aulario.py:64]: Type _: int_ right before a cast is pretty redundant

**Review by TendTo — CHANGES_REQUESTED** [review]

File: data/translations/en.yaml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+# /help command
+HELP_HEADER_TEXT_ID: "@DMI_Bot can help you with (following commands):"
+HELP_DIPARTIMENTO_CDL_KEYBOARD_TEXT_ID: "🏢 Department  and Degree Program"
+HELP_REGOLAMENTO_DIDATTICO_KEYBOARD_TEXT_ID: "🪧 Educational Regulations"
+HELP_SEGRETERIA_CONTATTI_KEYBOARD_TEXT_ID: "🕐 Receptionist's contacts and working hours"
+HELP_ERSU_ORARI_KEYBOARD_TEXT_ID: "🍽 ERSU's contactsand working hours"
+HELP_APPUNTI_CLOUD_KEYBOARD_TEXT_ID: "☁ Notes & Cloud"
+HELP_PROGETTI_RICONOSCIMENTI_KEYBOARD_TEXT_ID: "🏅 Projects e Credits"
+HELP_ALL_COMMANDS_KEYBOARD_TEXT_ID: "All commands"
+HELP_ALL_COMMANDS_TOOLTIP_ID: "
```
[the remark this answers is not in the document]
Comment (LightDestory) [reply data/translations/en.yaml]: Removed all double quotes for multi-line texts

Regarding the single lines it makes the code highlighting easier on IntelliJ

**Review by TendTo — APPROVED** [review]

**Review by drendog — APPROVED** [review]

File: module/data/vars.py [hunk 6]
```diff
@@ -116,8 +128,42 @@ class TEXT_IDS(Enum):
...
 PLACE_HOLDER: str = "%capybara%"
 
+
+# List of textx that are dynamic
+# The reverse is used as a translation table to bridge md and informative handlers into localization handler
+ON_DEMAND_TEXTS: dict[str, str] = {
+    "HELP_CDL_EXAMS_LINK_TOOLTIP_ID": "esami_link",
+    "HELP_CDL_CLASSES_LINK_TOOLTIP_ID": "lezioni_link",
+    "HELP_CDL_LIBRARY_TOOLTIP_ID": "biblioteca",
+    "REPRS_DMI_TOOLTIP_ID": "rappresentanti_dmi",
```
Comment (TendTo) [inline module/data/vars.py:141]: Why put this in "ON_DEMAND_TEXTS"? So it can easily be modified from the .md file?

**Review by TendTo — APPROVED** [review]

File: module/data/vars.py (hunk 6 above)
Comment (LightDestory) [reply module/data/vars.py:141]: md files contains only essential stuff like working hours or contacts, the message template with fixed sentences can be found inside yaml files.

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
We can merge for me 👍

**Comment by drendog** [comment]
bump

**Pull request merged**